### PR TITLE
Added button directive

### PR DESCRIPTION
**Pull request by ewelina-slepko**

### CODE REVIEW

File: src/app/authentication/authentication.module.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { AuthenticationComponent } from './authentication.component';
 import {AuthenticationRoutingModule} from './authentication-routing.module';
 import {FormsModule} from '@angular/forms';
 import { VerifyEmailComponent } from './verify-email/verify-email.component';
+import {DirectivesModule} from '../shared/directives/directives.module';
```
Comment (Rementh) [inline src/app/authentication/authentication.module.ts]: Format it

File: src/app/authentication/sign-in/sign-in.component.html [hunk 2]
```diff
@@ -14,9 +14,13 @@
...
-  <button type="submit">Sign In</button>
+  <button appSubmit type="submit">Sign In</button>
 </form>
 
-<button type="button" (click)="logInWithGoogle()">Log In with Google</button>
+<button (click)="signInWithGoogle()">Sign In with Google</button>
+
 <button type="button" routerLink="../signup">Don't have an account?</button>
+<button type="button" routerLink="/dashboard">Kliknij sobie</button>
+<button type="button" (click)="signOut()">Log out</button>
```
Comment (Rementh) [inline src/app/authentication/sign-in/sign-in.component.html]: No need to specify type="button" for element outside of the form

File: src/app/authentication/sign-in/sign-in.component.ts [hunk 3]
```diff
@@ -10,10 +11,22 @@ export class SignInComponent implements OnInit {
 
   user: User = {} as User;
 
-  constructor() {
+  constructor(public authService: AuthenticationService) {
```
Comment (Rementh) [inline src/app/authentication/sign-in/sign-in.component.ts]: Should be private if not used in the template

File: src/app/shared/services/authentication.service.ts [hunk 4]
```diff
@@ -4,18 +4,73 @@ import 'firebase/analytics';
...
+    firebase.auth().createUserWithEmailAndPassword(email, password).catch((error) => {
+      if (error) {
+        this.notifierService.notify(
+          error.message,
+          2,
+        );
+      }
+    }).then(res => {
+      if (res) {
+        this.redirectToDashboard();
```
Comment (Rementh) [inline src/app/shared/services/authentication.service.ts]: Auth service should be responsible only for authentication, not redirect. Consider moving redirect logic into signup screen.

File: src/app/shared/services/authentication.service.ts [hunk 5]
```diff
@@ -4,18 +4,73 @@ import 'firebase/analytics';
...
+      }
+    });
+
+  }
+
+  signInWithEmailAndPassword(email, password) {
+    firebase.auth().signInWithEmailAndPassword(email, password)
+      .catch((error) => console.log(error))
+      .then(res => {
+        this.redirectToDashboard();
```
Comment (Rementh) [inline src/app/shared/services/authentication.service.ts]: The same as before.

File: src/app/shared/services/authentication.service.ts [hunk 6]
```diff
@@ -4,18 +4,73 @@ import 'firebase/analytics';
...
+    firebase.auth().signOut().then(res => console.log(res));
+  }
+
+  isAuthenticated(): boolean {
+    const currentUser = firebase.auth().currentUser;
+    return !!currentUser;
+  }
 
+  redirectToDashboard() {
+    firebase.auth().onAuthStateChanged((user) => {
```
Comment (Rementh) [inline src/app/shared/services/authentication.service.ts]: The "onAuthStateChanged" listener is not needed here. Function is being used only in places where user is already present (after successful signin/signup), so there is no reason to check it again.

File: src/app/authentication/authentication.module.ts (hunk 1 above)
Comment (ewelina-slepko) [reply src/app/authentication/authentication.module.ts]: ok.

File: src/app/authentication/sign-in/sign-in.component.html (hunk 2 above)
Comment (ewelina-slepko) [reply src/app/authentication/sign-in/sign-in.component.html]: ok.

File: src/app/authentication/sign-in/sign-in.component.ts (hunk 3 above)
Comment (ewelina-slepko) [reply src/app/authentication/sign-in/sign-in.component.ts]: ok.

**Comment by ewelina-slepko** [comment]
Done.

**Comment by Rementh** [comment]
Looks good

**Pull request merged**